lib/logstash/outputs/s3.rb

```diff
@@ -439,4 +440,19 @@ def get_under_recovery_files(files)
     end
     skip_files
   end
+
+  # Sorts explored files in desc order
+  # And proceeds bottom-up traversal remove operation if dir is empty
+  # Returns removed set to avoid duplicate Dir.glob process
+  def remove_empty_dirs(files)
+    removed_dirs = Set.new
+    files.sort { |x, y| y.length - x.length } # make sure to remove from leaf node
+         .select { |path| ::File.directory?(path) }
+         .select { |path| (Dir.entries(path) - %w[ . .. ]).empty? } # current and parent dirs escape
+         .each do |path|
+            FileUtils.rm_rf(path, :secure => true)
+            removed_dirs << path
```

Comment:
Should we add a debug statement here that we are deleting a directory?

Reply:
Yup, good point.